### PR TITLE
Sca wrapper

### PR DESCRIPTION
**Pull request by TheDeadlyPianist**
# Pull Request Checklist
- Add/update Welsh translation if required.
- Add any logging needed for Bridge Team monitoring.
- Add unit/acceptance/performance tests for new features.
- Make sure code coverage has not dropped.
- Run the [cysp-acceptance-test](https://github.com/hmrc/cysp-acceptance-test).
- Run the [nisp-sapeak-performance-tests](https://github.com/hmrc/nisp-sapeak-performance-tests).
- Perform relevant accessibility checks.
- Test manually in browser if required.
- Make any environment configuration changes/checks if needed.

### CODE REVIEW

File: app/uk/gov/hmrc/nisp/views/Main.scala [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+      case Failure(exception)                       => throw exception
+    }
+
+    if (scaWrapperToggle.isEnabled) {
+      logger.debug(s"SCA Wrapper layout used for request `${request.uri}``")
+
+      wrapperService.layout(
+        content = contentBlock,
+        pageTitle = Some(fullPageTitle),
+        //serviceNameKey = appConfig.serviceNameKey,
```
Comment (pascal22p) [inline app/uk/gov/hmrc/nisp/views/Main.scala:95]: You are missing this parameter. If you set it in application.conf you don't have welsh.
serviceNameKey = Some(messages("label.service_name"))

The sca template already uses mesasages so you just pass "label.service_name" but I find this very confusing.

File: conf/application.conf [hunk 2]
```diff
@@ -250,3 +252,22 @@ hmrc-timeout-dialog {
...
+      referrer.url = "/check-your-state-pension"
+      redirect.url = "check-your-state-pension"
+    }
+    service-name {
+      messages-key = "Check or update your medical insurance benefit"
+    }
+    signout {
+      url = "/check-your-state-pension/sign-out"
+      alternative-url = null
+    }
```
Comment (pascal22p) [inline conf/application.conf:271]: Is it not already in the template as an argument?

I have the same issue in benefits-frontend so I don't remember if it was intended or just a mistake.

File: project/AppDependencies.scala [hunk 3]
```diff
@@ -29,11 +29,12 @@ object AppDependencies {
     "uk.gov.hmrc"       %% "http-caching-client"                % s"10.0.0-$playVersion",
     "uk.gov.hmrc"       %% "tax-year"                           % "3.0.0",
     "uk.gov.hmrc"       %% "play-conditional-form-mapping"      % s"1.12.0-$playVersion",
-    "uk.gov.hmrc"       %% "play-frontend-hmrc"                 % s"6.7.0-$playVersion",
     "uk.gov.hmrc"       %% "play-partials"                      % s"8.3.0-$playVersion",
     "uk.gov.hmrc"       %% "play-frontend-pta"                  % "0.4.0",
     "uk.gov.hmrc"       %% s"internal-auth-client-$playVersion" % "1.2.0",
     "uk.gov.hmrc.mongo" %% s"hmrc-mongo-$playVersion"           % hmrcMongoVersion,
+    "uk.gov.hmrc"       %% "mongo-feature-toggles-client"       % "0.2.0",
+    "uk.gov.hmrc"       %% "sca-wrapper"                        % "1.0.40",
```
Comment (pascal22p) [inline project/AppDependencies.scala:37]: 1.0.41 now...

**Review by pascal22p — CHANGES_REQUESTED** [review]

File: project/AppDependencies.scala (hunk 3 above)
Comment (TheDeadlyPianist) [reply project/AppDependencies.scala:37]: big sad

File: app/uk/gov/hmrc/nisp/views/Main.scala (hunk 1 above)
Comment (TheDeadlyPianist) [reply app/uk/gov/hmrc/nisp/views/Main.scala:95]: I don't know how I missed this during my regression or manual testing. The page header is just wrong when in the SCA Wrapper mode. Glad you caught this.

**Review by pascal22p — DISMISSED** [review]

File: app/uk/gov/hmrc/nisp/models/admin/FeatureFlags.scala [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.nisp.models.admin
+
+import uk.gov.hmrc.mongoFeatureToggles.model.FeatureFlagName
+
+case object SCAWrapperToggle extends FeatureFlagName {
+  override val name: String = "SCAWrapperToggle"
```
Comment (lewey) [inline app/uk/gov/hmrc/nisp/models/admin/FeatureFlags.scala:22]: Is there a naming convention for the toggle, or does it not matter?

File: app/uk/gov/hmrc/nisp/views/Main.scala [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+        //serviceNameUrl = None,
+        sidebarContent = sidebar,
+        signoutUrl = routes.StatePensionController.signOut.url,
+        //timeOutUrl = appConfig.timeOutUrl,
```
Comment (lewey) [inline app/uk/gov/hmrc/nisp/views/Main.scala:99]: Commented code

File: app/uk/gov/hmrc/nisp/views/oldMain.scala.html [hunk 6]
```diff
@@ -108,12 +108,12 @@
 
 
 @hmrcLayout(
-  pageTitle = Some(if(extendedTitle){fullTitle}else{pageTitle}),
+  pageTitle = Some(if(extendedTitle) {fullTitle} else {pageTitle}),
   serviceName = Some(messages("nisp.title")),
   serviceUrl = Some(appConfig.serviceUrl),
-  nonce = CSPNonce.get,
+//  nonce = CSPNonce.get,
```
Comment (lewey) [inline app/uk/gov/hmrc/nisp/views/oldMain.scala.html:114]: commented code

File: conf/application.conf [hunk 7]
```diff
@@ -250,3 +252,22 @@ hmrc-timeout-dialog {
...
+    future-timeout = 5
+    host = "http://localhost:9514" # used in feedback url
+    feedback-service-name = "CYSP/pension"
+    exit-survey-origin = "CYSP"
+    accessibility-statement {
+      referrer.url = "/check-your-state-pension"
+      redirect.url = "check-your-state-pension"
+    }
+    service-name {
+      messages-key = "Check or update your medical insurance benefit"
```
Comment (lewey) [inline conf/application.conf:266]: Should this be a message key and is this value correct?

**Review by lewey — CHANGES_REQUESTED** [review]

File: conf/application.conf (hunk 7 above)
Comment (TheDeadlyPianist) [reply conf/application.conf]: It has been removed.

File: app/uk/gov/hmrc/nisp/models/admin/FeatureFlags.scala (hunk 4 above)
Comment (TheDeadlyPianist) [reply app/uk/gov/hmrc/nisp/models/admin/FeatureFlags.scala:22]: I've never seen a naming convention for these toggles.

File: conf/application.conf [hunk 8]
```diff
@@ -250,3 +253,15 @@ hmrc-timeout-dialog {
   enableSynchroniseTabs = true
   defaultCountdownInSeconds = 120
 }
```
Comment (lewey) [inline conf/application.conf:255]: The property "contact-frontend.serviceId" looks to be missing. This allows the link "Is this page not working properly? (opens in new tab)" to be displayed

**Review by lewey — APPROVED** [review]

**Pull request merged**